### DISCUSSION TITLE
jsconfig.json

### DISCUSSION THREAD

**Discussion by zaceno** [community/community#19653] [Codespaces]
The typescript engine of the web-editor (the one you get when you press `.`) does not seem to pick up on the jsconfig.json in my root folder (which regular vs code on the desktop does).  

Is that a known limitation? Or is there a way to make it pick it up. 

Example repo: https://github.com/zaceno/arithmentist

EDIT: Clarifying: The way I can tell it isn't picking up the jsconfig.json file is that it does not understand any of my type imports, hence the type info on hover says all types are `any`. And there are no error-squigglies for incorrect type usage.

**Comment by KartikM123** [comment]
Hi @zaceno thanks for raising this issue!

Unfortunately, we currently don't have virtual file system support for Javascript on github.dev (we do have support on Codespaces).  Typescript support for github.dev is currently in progress, so these kinds of issues are on our radar!

Thanks again for raising this issue and let me know if you have any more questions!

**Reply by aMelonRind** [reply]
> Typescript support for github.dev is currently in progress,

How's it going? any updates?
Currently vscode.dev does have project-wide typescript server support (more specifically, [in a github repo](https://vscode.dev/github/aMelonRind/JsMacros-scripts/blob/main/types/playground.js)), so I kind of expected that [github.dev](https://github.dev/aMelonRind/JsMacros-scripts/blob/main/types/playground.js) can do it as well. But apparently it doesn't support it yet.

Edit: It seems like the typescript server works on github.dev now. nice!
